### DISCUSSION TITLE
issue: isInvalid state only changing after submit

### DISCUSSION THREAD

**Discussion by zhondori** [react-hook-form/react-hook-form#11591] [Q&A]
### Version Number

^7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/p/devbox/hookform-issue-shygwv

### Steps to reproduce

1. Type number less than 1334 in the input. And click submit button. You get validation error
2. Type number greater than 1334, then error message disappears, but border-color of input stays red, invalid.
3. Then click submit button. After that border-color changes to default color.

### Expected behaviour

The `isInvalid` state should update after typing. I tried `revalidateMode`, didn't work.

I saw error object that it has a `type: "validate"` but no message. I think that problem is because of `validate` function

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
I don't think I'm following; your reproduction steps describe revalidation behaviour on value change, whereas your codesandbox contains a form that revalidates on submission instead.

**Reply by zhondori** [reply]
Here is the reproduction video: https://www.veed.io/view/4da3f1ca-832c-4da3-aa69-48591620543c?panel=share

**Reply by Moshyfawn** [reply]
Do you press enter to submit the form when the first validation appears? I still don't quite follow.

**Reply by zhondori** [reply]
Yes, i pressed enter to submit
